rpcs3/Emu/Cell/lv2/sys_usbd.h

```diff
@@ -1,64 +1,403 @@
+enum : u8
+{
+	USB_DESCRIPTOR_DEVICE       = 0x01,
+	USB_DESCRIPTOR_CONFIG       = 0x02,
+	USB_DESCRIPTOR_STRING       = 0x03,
+	USB_DESCRIPTOR_INTERFACE    = 0x04,
+	USB_DESCRIPTOR_ENDPOINT     = 0x05,
+	USB_DESCRIPTOR_HID          = 0x21,
+	USB_DESCRIPTOR_ACI          = 0x24,
+	USB_DESCRIPTOR_ENDPOINT_ASI = 0x25,
+};
+
+struct UsbDeviceDescriptor
+{
+	le_t<u16,1> bcdUSB;
+	u8 bDeviceClass;
+	u8 bDeviceSubClass;
+	u8 bDeviceProtocol;
+	u8 bMaxPacketSize0;
+	le_t<u16,1> idVendor;
+	le_t<u16,1> idProduct;
+	le_t<u16,1> bcdDevice;
+	u8 iManufacturer;
+	u8 iProduct;
+	u8 iSerialNumber;
+	u8 bNumConfigurations;
+};
+
+struct UsbDeviceConfiguration
+{
+	le_t<u16,1> wTotalLength;
+	u8 bNumInterfaces;
+	u8 bConfigurationValue;
+	u8 iConfiguration;
+	u8 bmAttributes;
+	u8 bMaxPower;
+};
+
+struct UsbDeviceInterface
+{
+	u8 bInterfaceNumber;
+	u8 bAlternateSetting;
+	u8 bNumEndpoints;
+	u8 bInterfaceClass;
+	u8 bInterfaceSubClass;
+	u8 bInterfaceProtocol;
+	u8 iInterface;
+};
+
+struct UsbDeviceEndpoint
+{
+	u8 bEndpointAddress;
+	u8 bmAttributes;
+	le_t<u16,1> wMaxPacketSize;
+	u8 bInterval;
+};
+
+struct UsbDeviceEndpointAudio
+{
+	u8 bEndpointAddress;
+	u8 bmAttributes;
+	le_t<u16,1> wMaxPacketSize;
+	u8 bInterval;
+	u8 bRefresh;
+	u8 bSynchAddress;
+};
+
+struct UsbDeviceHID
+{
+	le_t<u16,1> bcdHID;
+	u8 bCountryCode;
+	u8 bNumDescriptors;
+	u8 bDescriptorType;
+	le_t<u16,1> wDescriptorLength;
+};
+
+struct UsbDeviceACInterfaceHeader
+{
+	u8 bDescriptorSubtype;
+	le_t<u16,1> bcdADC;
+	le_t<u16,1> wTotalLength;
+	u8 bInCollection;
+	u8 baInterfaceNr;
+};
+
+struct UsbDeviceACInputTerminal
+{
+	u8 bDescriptorSubtype;
+	u8 bTerminalID;
+	le_t<u16,1> wTerminalType;
+	u8 bAssocTerminal;
+	u8 bNrChannels;
+	le_t<u16,1> wChannelConfig;
+	u8 iChannelNames;
+	u8 iTerminal;
+};
+
+struct UsbDeviceACOutputTerminal
+{
+	u8 bDescriptorSubtype;
+	u8 bTerminalID;
+	le_t<u16,1> wTerminalType;
+	u8 bAssocTerminal;
+	u8 bSourceID;
+	u8 iTerminal;
+};
+
+struct UsbDeviceACFeatureUnit
+{
+	u8 bDescriptorSubtype;
+	u8 bUnitID;
+	u8 bSourceID;
+	u8 bControlSize;
+	u8 Channel0;
+	u8 Channel1;
+	u8 Channel2;
+	u8 iFeature;
+};
+
+struct UsbDeviceASInterface
+{
+	u8 bDescriptorSubtype;
+	u8 bTerminalLink;
+	u8 bDelay;
+	u8 wFormatTag;
+};
+
+struct UsbDeviceASFormatType1
+{
+	u8 bDescriptorSubtype;
+	u8 bFormatType;
+	u8 bNrChannels;
+	u8 bSubframeSize;
+	u8 bBitResolution;
+	u8 bSamFreqType;
+	le_t<u16,1> tSamFreq1;
+	u8 tSamFreq1High;
+	le_t<u16,1> tSamFreq2;
+	u8 tSamFreq2High;
+	le_t<u16,1> tSamFreq3;
+	u8 tSamFreq3High;
+	le_t<u16,1> tSamFreq4;
+	u8 tSamFreq4High;
+	le_t<u16,1> tSamFreq5;
+	u8 tSamFreq5High;
+};
+
+struct UsbDeviceASIsochronousDataEndpoint
+{
+	u8 bDescriptorSubtype;
+	u8 bmAttributes;
+	u8 bLockDelayUnits;
+	le_t<u16,1> wLockDelay;
+};
+
+// PS3 internal codes
+enum PS3StandardUsbErrors : u32
+{
+	HC_CC_NOERR = 0x00,
+	EHCI_CC_MISSMF = 0x10,
+	EHCI_CC_XACT = 0x20,
+	EHCI_CC_BABBLE = 0x30,
+	EHCI_CC_DATABUF = 0x40,
+	EHCI_CC_HALTED = 0x50,
```

Comment:
These values already exist there: https://github.com/RPCS3/rpcs3/blob/12dc3c18723788024fd62a8dd6dd0f24d830d479/rpcs3/Emu/Cell/Modules/cellUsbd.h#L31-L36
Can we avoid duplicate content in some way?